latex/maketables.py

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/env python2
+import os, sys
+import json
+
+bench_names = {
+    "java15lua5_3": "JavaLua",
+    "java15php": "JavaPHP",
+    "java15sqlite": "JavaSQL",
+    "lua5_3java15": "LuaJava",
+    "lua5_3php": "LuaPHP",
+    "lua5_3sqlite": "LuaSQL",
+    "phplua5_3": "PHPLua",
+    "phpjava15": "PHPJava",
+    "phpsqlite": "PHPSQL",
+    "sqlitejava15": "SQLJava",
+    "sqlitelua5_3": "SQLLua",
+    "sqlitephp": "SQLPHP",
+}
+
+heu_names = {
+    "all": "All",
+    "hist": "Parse tree",
+    "stack": "Stack",
+    "line": "Line",
+}
+
+EXPERIMENT_DIR = "../logs_paper/"
+
+class TableMaker:
+
+    def __init__(self, folders):
+        self.makros = []
+        self.corpus = {}
+        self.get_benchmarks(folders[0])
+        rows = []
+        for folder in folders:
+            data, corpus = self.parse(folder)
+            self.corpus[folder] = corpus
+            rows.append(self.make_row(folder, data))
+        table = self.make_table(rows)
+        print(table)
+        # add makros
+        with open("experimentstats.tex", "a") as f:
+            f.write("\n".join(self.makros))
+            f.write("\n")
+
+    def parse(self, heuristic):
+        data = {}
+        corpus = {}
+        for c in self.benchmarks:
+            totale = totalf = 0
+            # collect data from expression benchmarks
+            data[c], totale = self.collect_stats("{}/{}/{}".format(EXPERIMENT_DIR, heuristic, c))
+            if os.path.isdir("{}/func_{}/".format(EXPERIMENT_DIR, heuristic)):
+                # add data from function benchmarks
+                funcdata, totalf = self.collect_stats("{}/func_{}/{}".format(EXPERIMENT_DIR, heuristic, c))
+                for k in funcdata:
+                    data[c][k] = (data[c][k] + funcdata[k]) / 2
+            corpus[c] = totale + totalf
+        return data, corpus
+
+    def collect_stats(self, filename):
+        if not os.path.exists(filename):
```

Comment:
Should we error in this case rather than silently swallow the error?

Reply:
Not without changing other parts of the file too since the function benchmarks don't contain SQL compositions. It was easier to just return 0 in those cases, but I'm happy to change it.

Reply:
I'm not sure I quite understand but perhaps that does suggest that it's worth changing!